pkg/cmd/cli/cmd/debug.go

```diff
@@ -531,17 +532,24 @@ func (o *DebugOptions) transformPodForDebug(annotations map[string]string) (*kap
 		if pod.Labels == nil {
 			pod.Labels = make(map[string]string)
 		}
-		pod.Labels[debugPodLabelName] = pod.Name
 	} else {
-		pod.Labels = map[string]string{debugPodLabelName: pod.Name}
+		pod.Labels = map[string]string{}
 	}
 	// always clear the NodeName
 	pod.Spec.NodeName = o.NodeName
 
 	pod.ResourceVersion = ""
 	pod.Spec.RestartPolicy = kapi.RestartPolicyNever
-	// TODO: shorten segments, make incrementing?
-	pod.Name = fmt.Sprintf("%s-debug", pod.Name)
+	// shorten segments to handle long names and names with bad characters
```

Comment:
> Can you move this to a function in the generate libraries please?

You mean down here: https://github.com/kubernetes/kubernetes/blob/master/pkg/api/generate.go#L59 ?  How about I conform to the interface and upstream later if something else ever needs it?